### DISCUSSION TITLE
Small observation while setting SSID and password on ESP32 devices after switching to client mode - possible bug?

### DISCUSSION THREAD

**Discussion by hagenbuch** [arendst/Tasmota#19706] [Support request]
Just a minor issue but I could reproduce it on several boards with an ESP32 but not on ESP8266

I successfully flashed tasmota32.factory.bin on several Olimex ESP32-POE and on three Odroid Go (here with lvgl).

As we all know, Tasmota offers an Accesspoint at first boot, we connect to 192.168.4.1 and then enter the SSID and password of our local WLAN. Usually, Tasmota manages to even show us the new IP before then switching to Station mode.

This went also fine on the ESP32s - BUT:

Even when the device already had logged in to the new WLAN (Wifi) and I accessed it from within the LAN, I still was asked for SSID and password a second time. Then when I entered it a second time, all went fine.

I have no idea where this comes from but after all: ESP32 has two possible MAC adresses, on for WLAN and one for LAN. Maybe some code gets confused?

To be super precise: While I can't remember having had this on ESP8266 devices, I might never have used a factory binary there, so the problem might as well be in the "factory.bin" variant, being independent of the ESP32 or ESP8266 hardware.

Andreas

**Comment by Jason2866** [comment]
Which device are you using to configure Tasmota?

**Reply by hagenbuch** [reply]
The Olimex ESP32-POE and the Odroid Go I installed it onto (I hope I understood your question alright). I just used the website of the device.

**Reply by Jason2866** [reply]
Which device do you use for flashing and on boarding? The device where the captive portal opens.

**Reply by hagenbuch** [reply]
I flashed both (tasmota32.factory.bin) with esptool.py on Linux Mint but I did not add Wifi credentials in the flashing process. Same with the browser, it is Firefox 118.0.1 but same observation with Chromium 117.0.5938. Next time, I could try on a Mac..

**Reply by sfromis** [reply]
One possible cause of such weirdness is two devices with same config details, meaning that you may not be using the one you expect. Happens. Better verify the MAC address.

**Reply by hagenbuch** [reply]
Ok thanks. I was flashing and configuring multiple devices, one after the other. Never was more than one opening its Wifi AP at the same time. I'll try to look out for these things next time!

**Reply by Jason2866** [reply]
The device hardware where the config is made has an impact, "some" the esp chips dont like a lot.
This happens with every firmware, not Tasmota related.
The Esp32-C3 and Apple iPhone is a very bad combo. Needs several tries.
Most Android phones work at first try.

**Reply by Jason2866** [reply]
Just tried with my Odroid-Go. Worked always at first attempt.

**Reply by hagenbuch** [reply]
Alright.. I'll keep all that in mind :) - thank you.

It seems just a little strange to me because the same Tasmota device that sends the page to my browser while being already on a different IP than 192.168.4.1 still shows me the SSID dialog.
